### ISSUE TITLE
v5: change the default behaviour of carousels to make autoplay opt-in rather than opt-out

### ISSUE THREAD

**Issue by patrickhlauke** [twbs/bootstrap#32649]
After much head-scratching, I grokked that the carousel component currently:

- autoplays by default after the first interaction (which seems rather pointless/nonsensical)
- allows authors to make the autoplay start right on page load through the `data-bs-ride="carousel"` option
- lets authors optionally pause the carousel autplay when hovering over it
- lets authors suppress autoplay altogether (opt-out) by explicitly adding `data-bs-interval="false"`

I would argue that this component's behaviour is the exact opposite of what authors would expect (certainly coming from other frameworks, like AMP). The usual approach that, as an author, I'd expect is:

- carousel is static by default
- authors can explicitly opt into autoplaying at page load somehow (e.g. adding some sort of autoplay toggle, like `data-bs-autoplay`, defaulted to `false`, but authors can set it explicitly to `true` to opt-in)
- for autoplaying carousels, stop the autoplay after an interaction (i.e. the user has explicitly clicked somewhere, moved focus into the carousel, is hovering over it) by default (this respects the user's intent by default)
- allow authors to explicitly ignore users hovering over the carousel with the mouse (but don't do it for keyboard) - with added warning that the option should not be used for any kind of interactive slide content, as that will otherwise move the content from under the users' mouse pointer/feet. for this reason, don't offer this option for keyboard/focus, as there it would be even more annoying and potentially lead to focus being lost/reset to the start of the page. maybe call it `data-bs-hoverstop`, which defaults to `true`, but can be suppressed with `false`

admittedly this would now be a breaking change, but i think a change for the better.

**Comment by mdo** [comment]
Hmm unsure about making the breaking change, but I think I agree with your take. Carousels in general irk me, and things automatically happening without my interaction with them isn't the best. If we're concerned at all about the breaking change, I'm thinking it's a `v6` thing instead of a breaking change in `v5`.

**Comment by patrickhlauke** [comment]
fair enough, let's aim for v6. maybe for v5 it's worth adding a fairly explicit note somewhere up high in the documentation explaining that the current default behaviour is autoplaying after the first interaction (though it's completely silly for anything other than, don't know, image slideshows?), and highlighting very clearly that if this behaviour is not desired, authors should really add `data-bs-interval="false"`

**Comment by patrickhlauke** [comment]
and regarding accessibility: once a carousel starts and can't be paused/stopped (other than leaving your mouse over it, or keyboard focus on it) it essentially violates https://www.w3.org/TR/WCAG/#pause-stop-hide

**Comment by ffoodd** [comment]
@patrickhlauke couldn't we add a play/pause button as an optionnal element, like we do for controls and indicators? This is what I had in mind in #31908. Since it'd be an enhancement, this wouldn't be a BC. That won't help with the default behaviour, but at least we'd provide an easier way for users to disable sliding.

**Comment by uhafner** [comment]
I also noticed this that in v5 there is no way to suppress autoplay anymore. In v4 it was possible to disable autoplay by using something like `<div id="trend-carousel" class="carousel slide carousel-dark" data-bs-interval="false">`. Now this data attribute does not work anymore (tested with beta 3). Or is this a bug in v5?

**Comment by patrickhlauke** [comment]
just sticking a pin in this again that i'd definitely welcome the outlined changes when it comes to v6 ... the current implementation of carousel still feels completely unintuitive, backwards, and user-hostile by default to me

**Comment by patrickhlauke** [comment]
ooooh, nice one one @mdo
